tests/integration/slashing/keeper/slash_redelegation_test.go

```diff
@@ -185,3 +185,208 @@ func fundAccount(t *testing.T, ctx context.Context, bankKeeper bankkeeper.Keeper
+	_, err = stakingMsgServer.BeginRedelegate(ctx, redelMsg)
+	require.NoError(t, err)
+
+	undelegationPercentage := math.LegacyMustNewDecFromStr(undelegationPercentageStr)
+	undelegationAmountDec := math.LegacyNewDecFromInt(testCoins[0].Amount).Mul(undelegationPercentage)
+	amountToUndelegate := undelegationAmountDec.TruncateInt()
+
+	// next block
+	ctx, err = simtestutil.NextBlock(app, ctx, time.Duration(1))
+	require.NoError(t, err)
+
+	portionofTestCoins := sdk.NewCoins(sdk.NewCoin(bondDenom, amountToUndelegate))
+	fmt.Println("Undelegation amount:", portionofTestCoins[0].Amount)
+
+	undelMsg = stakingtypes.NewMsgUndelegate(testAcc3.String(), goodValAddr.String(), portionofTestCoins[0])
+	_, err = stakingMsgServer.Undelegate(ctx, undelMsg)
+	require.NoError(t, err)
+
+	// next block
+	ctx, err = simtestutil.NextBlock(app, ctx, time.Duration(1))
+	require.NoError(t, err)
+
+	// slash the evil val
+	err = slashKeeper.Slash(ctx, evilValConsAddr, math.LegacyMustNewDecFromStr(slashFraction), evilPower, misbehaveHeight)
+	require.NoError(t, err)
+
+	// assert invariants
+	_, stop := stakingkeeper.AllInvariants(stakingKeeper)(ctx)
+	require.False(t, stop)
+	_, stop = bankkeeper.AllInvariants(bankKeeper)(ctx)
+	require.False(t, stop)
+	_, stop = distributionkeeper.AllInvariants(distrKeeper)(ctx)
+	require.False(t, stop)
+
+	// fastforward few block (and time) to complete redelegations and unbondings
+	ctx, err = simtestutil.NextBlock(app, ctx, time.Duration(1000000000000000000))
+	require.NoError(t, err)
+	ctx, err = simtestutil.NextBlock(app, ctx, time.Duration(1000000000000000000))
+	require.NoError(t, err)
+	ctx, err = simtestutil.NextBlock(app, ctx, time.Duration(1000000000000000000))
+	require.NoError(t, err)
+
+	// we check all accounts should be slashed with the equal amount, and they should end up with same balance including staked amount
+	stakedAcc1, err := stakingKeeper.GetDelegatorBonded(ctx, testAcc1)
+	require.NoError(t, err)
+	stakedAcc2, err := stakingKeeper.GetDelegatorBonded(ctx, testAcc2)
+	require.NoError(t, err)
+	stakedAcc3, err := stakingKeeper.GetDelegatorBonded(ctx, testAcc3)
+	require.NoError(t, err)
+
+	balance1AfterSlashing := bankKeeper.GetBalance(ctx, testAcc1, bondDenom).Add(sdk.NewCoin(bondDenom, stakedAcc1))
+	balance2AfterSlashing := bankKeeper.GetBalance(ctx, testAcc2, bondDenom).Add(sdk.NewCoin(bondDenom, stakedAcc2))
+	balance3AfterSlashing := bankKeeper.GetBalance(ctx, testAcc3, bondDenom).Add(sdk.NewCoin(bondDenom, stakedAcc3))
+	fmt.Println("testAcc1 balance amount:", balance1AfterSlashing.Amount, "staked amount:", stakedAcc1)
+	fmt.Println("testAcc2 balance amount:", balance2AfterSlashing.Amount, "staked amount:", stakedAcc2)
+	fmt.Println("testAcc3 balance amount:", balance3AfterSlashing.Amount, "staked amount:", stakedAcc3)
+
+	require.Equal(t, balance1AfterSlashing.Amount, balance2AfterSlashing.Amount)
+	require.Equal(t, balance2AfterSlashing.Amount, balance3AfterSlashing.Amount)
+}
```

Comment:
Good test case. 
There some more scenarios with multiple re-/undelegations that are not covered, yet.  Maybe you can add them here, too.

For bonus points: 
* assert bondend and unbonded amounts separately to see how much of each was slashed
* cover validator status unbonded
* run into `sharesToUnbond.GT(delegation.Shares)`
* (personal preference) refactor to a table test
```
// setup account and validators
	specs := map[string]struct {
		slashFrac math.LegacyDec
		del      []math.Int
		redel    []math.Int
		undel    []math.Int
		expBonded  math.Int
		expUnbonded  math.Int
	}{
		"": {},
	}
	for name, spec := range specs {
		t.Run(name, func(t *testing.T) {
// run with cacheContext ...
		})
	}
```